avocado/utils/download.py

```diff
@@ -64,14 +65,23 @@ def url_download(url, filename, data=None, timeout=300):
     :param timeout: (optional) default timeout in seconds.
     :return: `None`.
     """
-    log.info('Fetching %s -> %s', url, filename)
+    def download():
+        log.info('Fetching %s -> %s', url, filename)
 
-    src_file = url_open(url, data=data, timeout=timeout)
-    try:
-        with open(filename, 'wb') as dest_file:
-            shutil.copyfileobj(src_file, dest_file)
-    finally:
-        src_file.close()
+        src_file = url_open(url, data=data)
+        try:
+            with open(filename, 'wb') as dest_file:
+                shutil.copyfileobj(src_file, dest_file)
```

Comment:
IIUC, this commits includes the time to write the file to disk under the timeout.  Before this commit, the timeout is already applied to the download part. This is technically correct, but I don't think it's worth all the extra complexity like starting a new process.

IMO it'd suffice to add another logging line to let the user know that the file was downloaded, so the timeout may not have been reached, but that writing to a file is now stuck while writing to disk.  Another way to look at this, is if we were to also attempt to handle failures to the process termination itself which could be in under uninterruptiple sleep because of the backing device state.

My overall point is that this goes beyond what we need to address, and may incur problems of its own.

Reply:
Comment to self: after reading the docstring changes in the next commit, and re-reading the `urllib` documentation, I now realise that only the establishment of the connection is covered by its built-in timeout parameter, and the transfer itself is covered under `shutil.copyfileobj()`, right?

Reply:
Yes. Right. That is the reason I'm adding the timeout to this method only.